### PR TITLE
Fix #5291(@wdio/allure-reporter) - return status failed when an assertion failed with expect-webdriverio

### LINKED ISSUE #5291: When assertion failed from expect-webdriverio, Test status is broken instead of failed in @wdio/allure-reporter
*Environment (please complete the following information):**
 - **WebdriverIO version:** 6.0.16
 - **Mode:** WDIO
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 12.6.0
 - **NPM version:** 6.9.0
 - **Browser name and version:** Any
 - **Platform name and version:**  Windows 10
 - **Additional wdio packages used (if applicable):** "@wdio/cli": "^6.0.16",
    "@wdio/cucumber-framework": "^6.0.16",
    "@wdio/dot-reporter": "^6.0.16",
    "@wdio/local-runner": "^6.0.16",
    "@wdio/mocha-framework": "^6.0.16",
    "@wdio/selenium-standalone-service": "^6.0.16",
    "@wdio/spec-reporter": "^6.0.16",
    "@wdio/sync": "^6.0.16",
    "@wdio/allure-reporter": "^6.0.14",
    "@wdio/sync": "^6.0.16",
     "expect": "^25.3.0",
    "expect-webdriverio": "^1.0.0",

**Describe the bug**
When assertion failed with expect-webdriverio, Test status is BROKEN in allure-report because expect pacakge throw an Error and not an AssertionError.  Allure report should be improved to taken in account this new Error.
**To Reproduce**

```
Feature: Test
My description feature

Scenario: Test
  Given I fail test

```

```
Given(
  'I fail test', () => {
    expect(false).toBeEnabled();
  }
);
```

**Expected behavior**
Test status should be FAILED in allure xml report instead of BROKEN

**wdio.config file:**
```
const expect = require('expect');
require('expect-webdriverio').setOptions({ wait: 30000 });
 framework: 'cucumber',

  reporters: ['spec',
    ['allure', {
      outputDir: 'report/allure-results',
      disableWebdriverStepsReporting: false,
      disableWebdriverScreenshotsReporting: false,
      useCucumberStepReporter: true,
    }]],
  cucumberOpts: {
    // <boolean> show full backtrace for errors
    backtrace: false,
    // <string[]> module used for processing required features
    requireModule: ['@babel/register'],
    // <boolean< Treat ambiguous definitions as errors
    failAmbiguousDefinitions: true,
    // <boolean> invoke formatters without executing steps
    // dryRun: false,
    // <boolean> abort the run on first failure
    failFast: false,
    // <boolean> Enable this config to treat undefined definitions as
    // warnings
    ignoreUndefinedDefinitions: false,
    // <string[]> ("extension:module") require files with the given
    // EXTENSION after requiring MODULE (repeatable)
    name: [],
    // <boolean> hide step definition snippets for pending steps
    snippets: true,
    // <boolean> hide source uris
    source: true,
    // <string[]> (name) specify the profile to use
    profile: [],
    // <string[]> (file/dir) require files before executing features
    require: [
      './src/steps/given.js',
      './src/steps/then.js',
      './src/steps/when.js',
      './src/steps/after.js',
      // Or search a (sub)folder for JS files with a wildcard
      // works since version 1.1 of the wdio-cucumber-framework
      // './src/**/*.js',
    ],
    // <string> specify a custom snippet syntax
    snippetSyntax: undefined,
    // <boolean> fail if there are any undefined or pending steps
    strict: true,
    // <string> (expression) only execute the features or scenarios with
    // tags matching the expression, see
    // https://docs.cucumber.io/tag-expressions/
    tagExpression: 'not @Pending',
    // <boolean> add cucumber tags to feature or scenario name
    tagsInTitle: true,
    // <number> timeout for step definitions
    timeout: 99999999,
  },
```

**Log**
Content of testsuite xml file
```
<?xml version='1.0'?>
<ns2:test-suite xmlns:ns2='urn:model.allure.qatools.yandex.ru' start='1587302774479' stop='1587304082630'>
    <name>Test</name>
    <title>Test</title>
    <test-cases>
        <test-case start='1587302774636' status='broken' stop='1587304081256'>
            <name>Test</name>
            <title>Test</title>
            <labels>
                <label name='language' value='javascript'/>
                <label name='framework' value='wdio'/>
                <label name='thread' value='0-0'/>
            </labels>
            <parameters>
                <parameter kind='argument' name='browser' value='chrome'/>
            </parameters>
            <steps>
                <step start='1587302774645' status='broken' stop='1587304081255'>
                    <name>Given I fail test</name>
                    <title>Given I fail test</title>
                    <attachments/>
                    <steps/>
                </step>
            </steps>
            <failure>
                <message>Expect  to be enabled
                
                Expected: "enabled"
                Received: "not enabled"</message>
                <stack-trace>Error: Expect  to be enabled
                
                Expected: "enabled"
                Received: "not enabled"
                    at World.&lt;anonymous> (C:\Users\fcellier\Documents\02-Test_Auto\end-to-end-tests-wdio5\src\steps\/given.js:43:19)
                    at World.executeSync (C:\Users\fcellier\Documents\02-Test_Auto\end-to-end-tests-wdio5\node_modules\@wdio\sync\build\index.js:56:18)
                    at C:\Users\fcellier\Documents\02-Test_Auto\end-to-end-tests-wdio5\node_modules\@wdio\sync\build\index.js:82:70</stack-trace>
            </failure>
        </test-case>
    </test-cases>
</ns2:test-suite>
```

### PR DESCRIPTION
**Pull request by lacell75**
## Proposed changes

Fixed #5291 

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-allure-reporter/tests/__fixtures__/testState.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ export function testFailedWithMultipleErrors() {
     return Object.assign(testState(), { errors, state: 'failed', end: '2018-05-14T15:17:21.631Z', _duration: 2730 })
 }
 
-export function testFailedWithASCIIColorCode() {
+export function testFailedWithAssertionErrorFormExpectWebdriverIO() {
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/tests/__fixtures__/testState.js:50]:
```suggestion
export function testFailedWithAssertionErrorFromExpectWebdriverIO() {
```

File: packages/wdio-allure-reporter/src/utils.js [hunk 2]
```diff
@@ -14,13 +14,18 @@ export const getTestStatus = (test, config) => {
         return testStatuses.FAILED
     }
 
+    if (test.error.name && test.error.message) {
+        const message = test.error.message.trim()
+        return (test.error.name === 'AssertionError' || message.startsWith('Expect'))  ? testStatuses.FAILED : testStatuses.BROKEN
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/utils.js:19]: How certain are we that every message starts with `Expect`?

File: packages/wdio-allure-reporter/tests/suite.test.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ import { clean, getResults } from 'wdio-allure-helper'
 import AllureReporter from '../src/'
 import { runnerEnd, runnerStart } from './__fixtures__/runner'
 import { suiteEnd, suiteStart } from './__fixtures__/suite'
-import { testFailed, testPassed, testPending, testStart, testFailedWithMultipleErrors, testFailedWithASCIIColorCode } from './__fixtures__/testState'
+import { testFailed, testPassed, testPending, testStart, testFailedWithMultipleErrors, testFailedWithAssertionErrorFormExpectWebdriverIO } from './__fixtures__/testState'
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/tests/suite.test.js:13]:
```suggestion
import { testFailed, testPassed, testPending, testStart, testFailedWithMultipleErrors, testFailedWithAssertionErrorFromExpectWebdriverIO } from './__fixtures__/testState'
```

File: packages/wdio-allure-reporter/tests/suite.test.js [hunk 4]
```diff
@@ -211,7 +211,7 @@ describe('Failed tests', () => {
         reporter.onRunnerStart(runnerEvent)
         reporter.onSuiteStart(suiteStart())
         reporter.onTestStart(testStart())
-        reporter.onTestFail(testFailedWithASCIIColorCode())
+        reporter.onTestFail(testFailedWithAssertionErrorFormExpectWebdriverIO())
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/tests/suite.test.js:214]:
```suggestion
        reporter.onTestFail(testFailedWithAssertionErrorFromExpectWebdriverIO())
```

**Comment by lacell75** [comment]
> I wonder if we can embed something in the `expect-webdriverio` package to make this simpler.

To fix it in the `expect-webdriverio` package, it is not easy, you should use assert instead of expect because `expect` throw Error only

File: packages/wdio-allure-reporter/src/utils.js (hunk 2 above)
Comment (lacell75) [reply packages/wdio-allure-reporter/src/utils.js:19]: I'm sure that every message starts with `Expect` , please see code in the `expect-webdriverio` package :   https://github.com/webdriverio/expect-webdriverio/blob/d27f4a30f6b1866e8847ce9df26fa613d54e8ed6/src/util/formatMessage.ts#L89

**Pull request merged**